addons/website_slides/static/tests/tours/slides_tour_tools.js

```diff
@@ -87,6 +95,139 @@ var addArticleToSection = function (sectionName, pageName, backend) {
+	run: async () => {
+		const img = backend ? $('iframe').contents().find('.o_wslides_fs_player img') : $('.o_wslides_fs_player img');
+		if (await checkURLContent(img.attr('src'), 'Overview.png', 'image/png', testPngImage)) {
+			img.addClass('o_wslides_tour_success');
+		}
+	}
+}, {
+	content: 'eLearning: check uploaded image content',
+	trigger: (backend ? 'iframe ' : '') + '.o_wslides_fs_player img.o_wslides_tour_success',
+},
+{
+	content: 'eLearning: back to course',
+	trigger: (backend ? 'iframe ' : '') + '.o_wslides_fs_sidebar_header a:contains("Déboulonnate")',
+}];
+};
+
+const addPdfToSection = function (sectionName, pageName, backend) {
+	return [
+{
+	content: 'eLearning: add content to section',
+	trigger: (backend ? 'iframe ' : '') + 'div.o_wslides_slide_list_category_header:contains("' + sectionName + '") a:contains("Add Content")',
+}, {
+	content: 'eLearning: click on document',
+	trigger: (backend ? 'iframe ' : '') + 'a[data-slide-category=document]',
+}, {
+	content: 'eLearning: choose upload from device',
+	trigger: (backend ? 'iframe ' : '') + '#source_type_local_file',
+}, {
+    content: 'eLearning: load pdf',
+    trigger: 'body',
+    run: async () => {
+        const upload = backend ?
+			document.getElementsByTagName('iframe')[0].contentWindow.document.getElementById('upload') :
+			document.getElementById('upload');
+		await fillInFileInput(upload, 'Exercise.pdf', 'application/pdf', testPdf);
+    },
+}, {
+    content: 'eLearning: click on tags',
+    trigger: (backend ? 'iframe ' : '') + 'ul.select2-choices:first',
+}, {
+    content: 'eLearning: select Exercise tag',
+    trigger: (backend ? 'iframe ' : '') + 'div.select2-result-label:contains("Exercise")',
+    in_modal: false,
+}, {
+	content: 'eLearning: fill exercise completion time',
+	trigger: (backend ? 'iframe ' : '') + 'input[name=duration]',
+	run: 'text 10',
+}, {
+    content: 'eLearning: create and publish slide',
+    trigger: (backend ? 'iframe ' : '') + 'footer.modal-footer button:contains("Publish")',
+}, {
+    content: 'eLearning: launch content',
+    trigger: (backend ? 'iframe ' : '') + 'a.o_wslides_js_slides_list_slide_link:contains("Exercise")',
+}, {
+    content: 'eLearning: check uploaded pdf presence and perform comparison',
+	trigger: (backend ? 'iframe ' : '') + '.o_wslides_fs_content',
+	run: async () => {
+		const pdfViewer = backend ?
+			$('iframe').contents().find('iframe').contents().find('#PDFSlideViewer') :
+			$('iframe').contents().find('#PDFSlideViewer');
+		if (await checkURLContent(pdfViewer.attr('data-slideurl'), 'Exercise.pdf', 'application/pdf', testPdf)) {
+			const target = backend ? $('iframe').contents().find('.o_wslides_fs_content') : $('.o_wslides_fs_content');
+			target.addClass('o_wslides_tour_success'); // cannot target sub iframe in trigger
+		}
+	},
+}, {
+	content: 'eLearning: check uploaded pdf content',
+	trigger: (backend ? 'iframe ' : '') + '.o_wslides_fs_content.o_wslides_tour_success',
```

Comment:
Is it not dangerous to check the same class as the "image" part?

Or is the DOM re-initialized at some point which will make them not interfere with each other?